### PR TITLE
prepare 1.2.10

### PR DESCRIPTION
**Pull request by RomanIovlev**

### CODE REVIEW

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 2
           

Complexity decreasing per file
==============================
+ jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/DataTable.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697&bid=18282449)

**Comment by mf-code** [comment]
[5390] - H4DGVHZP

**Comment by mf-code** [comment]
[5390] - fYij77T3

**Comment by mf-code** [comment]
[5390] - Allure report on Netlify: https://5ecaf78c40c1629ac25dea22--musing-swartz-6fc060.netlify.app

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 9
           

Complexity decreasing per file
==============================
+ jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/DataTable.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697&bid=18282449)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java [hunk 1]
```diff
@@ -2,10 +2,11 @@
 
 import com.epam.jdi.light.asserts.core.SoftAssert;
 import com.epam.jdi.light.elements.base.JDIBase;
-import com.epam.jdi.light.elements.interfaces.base.IBaseElement;
-import com.epam.jdi.light.elements.interfaces.base.JDIElement;
+import com.epam.jdi.light.elements.interfaces.base.*;
 import com.epam.jdi.tools.func.JFunc1;
 
+import static com.epam.jdi.tools.ReflectionUtils.*;
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java:8]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.tools.ReflectionUtils'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/BaseTable.java [hunk 2]
```diff
@@ -233,9 +230,26 @@ public WebList webColumn(int colNum) {
...
+    }
+    public int jsRowIndexByName(String rowName) {
+        int index = getRowHeaderIndex();
+        if (index == -1)
+            index = 1;
+        List<String> rowHeader;
+        try {
+            rowHeader = jsColumn(index);
+            if (rowHeader.size() == 0)
+                throw new RuntimeException();
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/BaseTable.java:243]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java [hunk 3]
```diff
@@ -145,7 +144,7 @@ private boolean isDisplayed(T element) {
      */
     @JDIAction("Assert that '{name}' is empty")
     public DataListAssert<T, D> empty() {
-        jdiAssert(element.isEmpty() ? "list is empty" : "list is not empty", Matchers.is("list is empty"));
+        jdiAssert(element().isEmpty() ? "list is empty" : "list is not empty", Matchers.is("list is empty"));
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:147]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java [hunk 4]
```diff
@@ -155,7 +154,7 @@ private boolean isDisplayed(T element) {
      */
     @JDIAction("Assert that '{name}' is not empty")
     public DataListAssert<T, D> notEmpty() {
-        jdiAssert(element.isEmpty() ? "list is empty" : "list is not empty", Matchers.is("list is not empty"));
+        jdiAssert(element().isEmpty() ? "list is empty" : "list is not empty", Matchers.is("list is not empty"));
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:157]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/elements/complex/IList.java [hunk 5]
```diff
@@ -1,7 +1,6 @@
 package com.epam.jdi.light.elements.complex;
 
-import com.epam.jdi.light.elements.interfaces.base.HasValue;
-import com.epam.jdi.light.elements.interfaces.base.IBaseElement;
+import com.epam.jdi.light.elements.interfaces.base.*;
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/elements/complex/IList.java:3]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.elements.interfaces.base'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java [hunk 6]
```diff
@@ -110,7 +109,7 @@
     @JDIAction("Assert that '{name}' elements [{0}] are displayed")
     public DataListAssert<T, D> displayed(String... names) {
         for (String name : names)
-            jdiAssert(element.get(name).isDisplayed() ? name + "displayed" : "hidden", Matchers.is(name + "displayed"));
+            jdiAssert(element().get(name).isDisplayed() ? name + "displayed" : "hidden", Matchers.is(name + "displayed"));
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:112]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java [hunk 7]
```diff
@@ -120,7 +119,7 @@
      */
     @JDIAction("Assert that '{name}' has at least one displayed element")
     public DataListAssert<T, D> displayed() {
-        jdiAssert(element.isDisplayed() ? "displayed" : "hidden", Matchers.is("displayed"));
+        jdiAssert(element().isDisplayed() ? "displayed" : "hidden", Matchers.is("displayed"));
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:122]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java [hunk 8]
```diff
@@ -130,7 +129,7 @@
      */
     @JDIAction("Assert that '{name}' is hidden")
     public DataListAssert<T, D> hidden() {
-        jdiAssert(map(element, this::isDisplayed), everyItem(Matchers.is(false)));
+        jdiAssert(map(element(), this::isDisplayed), everyItem(Matchers.is(false)));
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:132]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java [hunk 9]
```diff
@@ -2,10 +2,11 @@
 
 import com.epam.jdi.light.asserts.core.SoftAssert;
 import com.epam.jdi.light.elements.base.JDIBase;
-import com.epam.jdi.light.elements.interfaces.base.IBaseElement;
-import com.epam.jdi.light.elements.interfaces.base.JDIElement;
+import com.epam.jdi.light.elements.interfaces.base.*;
```
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java:5]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.elements.interfaces.base'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

**Comment by mf-code** [comment]
[5409] - FQlpv5QE

**Comment by mf-code** [comment]
[5409] - 6K41P2mw

**Comment by mf-code** [comment]
[5409] - Allure report on Netlify: https://5ecc487451f1bc53b2610f8d--musing-swartz-6fc060.netlify.app

**Comment by codacy-review** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 7
- Added 9
           

Complexity decreasing per file
==============================
+ jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/DataTable.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697&bid=18282449)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java (hunk 7 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:122]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/BaseTable.java (hunk 2 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/elements/complex/table/BaseTable.java:243]: Codacy Issue found: [Avoid throwing raw exception types.](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java (hunk 8 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:132]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java (hunk 9 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java:5]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.elements.interfaces.base'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java (hunk 3 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:147]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java (hunk 4 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:157]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java (hunk 6 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/core/DataListAssert.java:112]: Codacy Issue found: [Unnecessary use of fully qualified name 'Matchers.is' due to existing static import 'org.hamcrest.Matchers.*'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/elements/complex/IList.java (hunk 5 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/elements/complex/IList.java:3]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.light.elements.interfaces.base'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

File: jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java (hunk 1 above)
Comment (codacy-review) [inline jdi-light/src/main/java/com/epam/jdi/light/asserts/generic/BaseAssert.java:8]: Codacy Issue found: [Avoid unused imports such as 'com.epam.jdi.tools.ReflectionUtils'](https://app.codacy.com/manual/42_/jdi-light/pullRequest?prid=5576697)

**Comment by mf-code** [comment]
[5410] - 2x6GsiHV

**Comment by mf-code** [comment]
[5410] - twBIJPQA

**Comment by mf-code** [comment]
[5410] - Allure report on Netlify: https://5ecc52f9566965b7383ccab9--musing-swartz-6fc060.netlify.app

**Pull request merged**